ts/src/pro/mexc.ts

```diff
@@ -361,26 +361,37 @@ export default class mexc extends mexcRest {
         /**
          * @method
          * @name mexc3#watchOrderBook
-         * @see https://mxcdevelop.github.io/apidocs/spot_v3_en/#diff-depth-stream
+         * @see https://mexcdevelop.github.io/apidocs/spot_v3_en/#diff-depth-stream
+         * @see https://mexcdevelop.github.io/apidocs/contract_v1_en/#depth
          * @description watches information on open orders with bid (buy) and ask (sell) prices, volumes and other data
          * @param {string} symbol unified symbol of the market to fetch the order book for
-         * @param {int} [limit] the maximum amount of order book entries to return
+         * @param {int} [limit] 5, 10, or 20
          * @param {object} [params] extra parameters specific to the exchange API endpoint
          * @returns {object} A dictionary of [order book structures]{@link https://docs.ccxt.com/#/?id=order-book-structure} indexed by market symbols
          */
         await this.loadMarkets ();
         const market = this.market (symbol);
         symbol = market['symbol'];
-        const messageHash = 'orderbook:' + symbol;
+        let messageHash = 'orderbook:' + symbol;
         let orderbook = undefined;
         if (market['spot']) {
-            const channel = 'spot@public.increase.depth.v3.api@' + market['id'];
+            let channel = 'spot@public.increase.depth.v3.api@' + market['id'];
+            if (limit !== undefined) {
+                if ((limit !== 5) && (limit !== 10) && (limit !== 20)) {
+                    throw new BadRequest (this.id + ' watchOrderBook () limit argument can only be 5, 10 or 20');
+                }
+                messageHash += ':' + limit;
```

Comment:
@samgermain this breaks in python, we must be careful with these things

```
 p mexc watchOrderBook "BTC/USDT" 5 > out.txt
/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py:124: SyntaxWarning: invalid escape sequence '\s'
  call_reg = "\s*(\w+)\s*\.\s*(\w+)\s*\(([^()]*)\)"
Traceback (most recent call last):
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 246, in <module>
    asyncio.run(main())
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 225, in main
    result = await result
             ^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/mexc.py", line 364, in watch_order_book
    messageHash += ':' + limit
                   ~~~~^~~~~~~
TypeError: can only concatenate str (not "int") to str
```

Reply:
also if we add the limit to the messageHash, then we would need to adjust the resolving as well